tests/h/admin/views/features_test.py

```diff
@@ -75,21 +77,58 @@ def test_cohorts_index_when_no_cohorts():
     assert result["results"] == []
 
 
-def test_new_cohort_creates_cohort():
+def test_new_cohort_creates_cohort_with_no_members():
     req = DummyRequest(db=db.Session)
     req.params['add'] = 'cohort'
     result = views.cohorts_add(req)
     assert isinstance(result, httpexceptions.HTTPSeeOther)
 
     result = views.cohorts_index({}, req)
     assert len(result["results"]) == 1
-    assert result["results"][0].name == "cohort"
+
+    cohort = result["results"][0]
+    assert cohort.name == "cohort"
+    assert len(cohort.members) == 0
+
+
+def create_cohort(req):
+    req.params['add'] = 'cohort'
+    views.cohorts_add(req)
+    return views.cohorts_index({}, req)['results'][0]
+
+
+def add_user_to_cohort(req, user, cohort):
+    req.matchdict['id'] = cohort.id
+    req.params['add'] = user.username
+    views.cohorts_edit_add(req)
+
+
+def test_cohort_edit_add_and_remove_user():
```

Comment:
And in case it's not clear _why_ tests "should test one thing at a time," it's so it's easy to identify what exactly is broken when a test fails. In this case, there are (at least) three different functions (not counting the use of `create_cohort` and by extension the `cohorts_add` and `cohorts_index` views) which could have broken, and I'd then have to drill into this test before I even understood what was broken.